x/mongo/driver/topology/topology_options.go

```diff
@@ -333,6 +335,28 @@ func NewConfig(co *options.ClientOptions, clock *session.ClusterClock) (*Config,
 		)
 	}
 
+	if opts := co.LoggerOptions; opts != nil {
+		if opts == nil {
+			opts = options.Logger()
+		}
```

Comment:
This `opts == nil` check will never be true because we just verified that `opts != nil` on the previous line.